### ISSUE TITLE
Non-Resource API Returns EventProxy with Dangling Resource Pointer

### ISSUE THREAD

**Issue by MrBurmark** [LLNL/RAJA#1048]
The non-resource API currently makes a copy of the default resource on the stack and then passes a reference to that copy into the resource version of the API. The resource version of the API passes that resource reference into the implementation which returns an EventProxy object holding a pointer to the resource. When this EventProxy is returned from the non-resource API the resource that it points to no longer exists. This can be fixed by taking a reference to the default resource instead of copying it.
```
EventProxy<RES> foo_resource(RES& r) {
  // do foo with r
  return EventProxy<RES>(&r);
}
EventProxy<RES> foo() {
  // This is what is currently in the code
  auto r = RES::get_default();
  // This is a possible fix
  // auto& r = RES::get_default();
  return foo_resource(r);
}
```

**Comment by mdavis36** [comment]
Good catch @MrBurmark I'll go through and put a fix up for this

**Comment by mdavis36** [comment]
Resolved in #1054
